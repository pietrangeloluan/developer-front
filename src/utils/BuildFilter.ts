export default function (params: any) {
  let filter = {}
  for (const key in params) {
    if (Object.hasOwnProperty.call(params, key)) {
      const parameter = params[key]
      if (parameter) {
        filter = { ...filter, [key]: parameter }
      }
    }
  }
  return filter
}
