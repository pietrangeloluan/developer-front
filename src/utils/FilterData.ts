export default function (obj, filter) {
  const filtered = obj.filter(item => {
    for (const key in filter) {
      if (item[key] === undefined || !filter[key].includes(item[key])) {
        return false
      }
    }
    return true
  })

  return filtered
}
