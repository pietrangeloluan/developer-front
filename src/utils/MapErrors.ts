export function mapErrors(errors, context) {
  const errorKeys = Object.keys(errors)
  const regex = new RegExp(`${context}\\[\\d+\\]`, 'g')
  const contextErrors = errorKeys.filter(err => err.match(regex))

  if (!contextErrors.length) return [{}]

  const updated = []

  contextErrors.forEach(err => {
    const found = [...err.matchAll(/\[(\d+)\]\.(\w+)/g)]

    if (found.length) {
      found[0].shift()
      const errIndex = found[0][0]
      const errField = found[0][1]

      updated[errIndex] = {
        ...updated[errIndex],
        [errField]: errors[`${context}[${errIndex}].${errField}`]
      }
    }
  })

  return updated
}
