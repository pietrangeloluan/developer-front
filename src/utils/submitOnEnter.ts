import { KeyboardEvent } from 'react'

export function submitOnEnter(fn: Function, args: any[] = []) {
  return function (e: KeyboardEvent) {
    if (e.key === 'Enter') fn.apply(args)
  }
}
