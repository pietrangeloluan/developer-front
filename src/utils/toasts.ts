import { toast } from 'react-toastify'

export const toasts = {
  created() {
    return toast.success('Dados cadastrados com sucesso.', {
      autoClose: 5 * 1000
    })
  },
  deleted() {
    return toast.success('Dados exclu√≠dos com sucesso.', {
      autoClose: 5 * 1000
    })
  },
  edited() {
    return toast.success('Dados editados com sucesso.', { autoClose: 5 * 1000 })
  },
  generalFail({ code = 'UNKNOWN_ERROR' } = {}) {
    return toast.error(
      'Ocorreu uma falha inesperada. Por favor, aguarde alguns minutos de tente novamente.',
      { autoClose: 5 * 1000, toastId: code }
    )
  }
}
