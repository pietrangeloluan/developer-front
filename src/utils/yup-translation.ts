/* eslint-disable no-template-curly-in-string */
import { setLocale } from 'yup'

const translation = {
  array: {
    max: '${path} deve conter no máximo ${max} itens',
    min: '${path} deve conter no mínimo ${min} itens'
  },
  date: {
    max: '${path} deve ser anterior à ${max}',
    min: '${path} deve ser mais recente que ${min}',
    required: '${path} é obrigatório'
  },
  mixed: {
    default: '${path} é inválido',
    notOneOf: '${path} não pode ser igual à um destes valores: ${values}',
    oneOf: '${path} deve ser igual à um dos seguintes valores: ${values}',
    required: '${path} é obrigatório'
  },
  number: {
    integer: '${path} deve ser inteiro',
    lessThan: '${path} deve ser menor à ${less}',
    max: '${path} deve ser no máximo ${max}',
    min: '${path} deve ser no mínimo ${min}',
    moreThan: '${path} deve ser maior à ${more}',
    negative: '${path} deve ser negativo',
    notEqual: '${path} não pode ser igual à ${notEqual}',
    positive: '${path} deve ser positivo',
    required: '${path} é obrigatório'
  },
  string: {
    email: '${path} não contém um e-mail válido',
    length: '${path} deve conter exatamente ${length} caracteres',
    lowercase: '${path} deve ser maiúsculo',
    max: '${path} deve conter no máximo ${max} caracteres',
    min: '${path} deve conter pelo menos ${min} caracteres',
    trim: '${path} não deve conter espaços em branco em início ou final.',
    uppercase: '${path} deve ser minúsculo',
    url: '${path} não contém uma URL válida'
  }
}

setLocale(translation)
