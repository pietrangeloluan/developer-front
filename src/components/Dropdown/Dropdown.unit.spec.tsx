import { fireEvent, render, screen } from '@testing-library/react'

import { renderWithThemeProvider } from '__tests__/utils'

import { Dropdown } from '.'

describe('<Dropdown />', () => {
  beforeEach(() => {
    const title = <h1 aria-label="toogle dropdown">Click here</h1>

    render(
      <Dropdown title={title}>
        <span>content</span>
      </Dropdown>
    )
  })

  it('should render title', () => {
    expect(screen.getByLabelText(/toogle dropdown/)).toBeInTheDocument()
  })

  it('should match the snapshot', () => {
    const title = <h1 aria-label="toogle dropdown">Click here</h1>
    const { container } = renderWithThemeProvider(Dropdown, {
      title
    })
    expect(container).toMatchSnapshot()
  })

  it('should handle open/close dropdown', () => {
    const content = screen.getByText(/content/).parentElement!

    expect(content).toHaveStyle({ opacity: 0 })
    expect(content.getAttribute('aria-hidden')).toBe('true')

    fireEvent.click(screen.getByLabelText(/toogle dropdown/))

    expect(content).toHaveStyle({ opacity: 1 })
    expect(content.getAttribute('aria-hidden')).toBe('false')
  })
})
