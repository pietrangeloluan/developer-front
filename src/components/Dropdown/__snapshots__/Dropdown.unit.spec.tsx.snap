// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should match the snapshot 1`] = `
.c1 {
  cursor: pointer;
  color: white;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 4;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: white;
  color: black;
  margin-top: 0.6rem;
  position: absolute;
  right: 0;
  z-index: 5;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c0 .c2,
.c0 .sc-dlnjwi {
  border-radius: 0.8rem;
  position: absolute;
  right: 4.8rem;
  -webkit-transition: -webkit-transform 0.2s ease-in,opacity 0.3s ease-in-out;
  -webkit-transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  z-index: 200;
  opacity: 0;
  pointer-events: none;
  -webkit-transform: translateY(-2rem);
  -ms-transform: translateY(-2rem);
  transform: translateY(-2rem);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        aria-label="toogle dropdown"
      >
        Click here
      </h1>
    </div>
    <div
      aria-hidden="true"
      class="c2 c3"
    />
  </div>
</div>
`;
